code/modules/games/cards.dm

```diff
@@ -214,7 +224,7 @@
 	if(!discarding || !to_discard[discarding] || !usr || !src) return
 
 	var/datum/playingcard/card = to_discard[discarding]
-	qdel(to_discard)
+	to_discard.Cut()
```

Comment:
Warning, tech talk ahead.
Lists should never be qdel()d, because the garbage collector can only handle subtypes of /datum which lists are not.
There are a few places where it happens due to my mass change of del() -> qdel(), and I should've checked for it.

Nulling a list is generally a bad idea unless BYOND has fixed a reference leakage bug. Cut() (and previously del()) is definitely preferable. It's just that some pieces of code check for null-lists to do initialization and similar sort of things.